plugin/pkg/scheduler/api/v1/types.go

```diff
@@ -158,8 +158,18 @@ type ExtenderArgs struct {
 	NodeNames *[]string `json:"nodenames,omitempty"`
 }
 
-// FailedNodesMap represents the filtered out nodes, with node names and failure messages
-type FailedNodesMap map[string]string
```

Comment:
This change is not backward compatible. We would break existing extenders if we add a new field to the response that scheduler expects from extenders. We should handle a response that does not use `FailureInfo`. We should make a FailureInfo out of a string that an existing extender may send back.

Reply:
In that case the we have to assume extender send back string like `"PodFitsHostPorts#IsUnResolvableByPreemption"` and then split in scheduler ... does this make sense?  (kind of ugly somehow ...)

Reply:
Yes, something along this. API code probably has functions to marshal/unmarshal such data, but I agree with you that this is a bit ugly. I wonder if this change is necessary to address #51656. I guess we can leave the code as before, i.e., marking predicates as resolvable or unresolvable in a hardcoded way and assume all the new failure messages by extenders as resolvable. We just add the mechanism of sending nodes and victims in bulk to the extenders.

Reply:
For the response part, in long term I would still like to see this change happen, hacky (buggy) string response is the last thing I want to see ...

While in short term, I agree with you that we can assume all extenders' scheduling failures are resolvable. As I can see the current preemption logic only assumes scheduling will success next time, so this assumption will not break the normal logic. I will send a new one.